CRM/Sepa/Logic/Batching.php

```diff
@@ -482,6 +480,57 @@ static function closeEnded() {
    **                                                                        **
    ****************************************************************************/
 
+  public static function getOrCreateTransactionGroup(
+    int $creditor_id,
+    string $mode,
+    string $collection_date,
+    ?int $financial_type_id,
+    int $notice,
+    array &$existing_groups
+  ): int {
+    $group_status_id_open = (int) CRM_Core_PseudoConstant::getKey('CRM_Batch_BAO_Batch', 'status_id', 'Open');
+
+    if (!isset($existing_groups[$collection_date][$financial_type_id ?? 0])) {
+      // this group does not yet exist -> create
+
+      // find unused reference
+      $reference = self::getTransactionGroupReference($creditor_id, $mode, $collection_date, $financial_type_id);
+
+      $group = civicrm_api('SepaTransactionGroup', 'create', array(
```

Comment:
`civicrm_api()` is deprecated. Should be replaced by `civicrm_api3()` here, or preferable replaced by APIv4.

Reply:
Correct, but there are 72 usages in the entire extension that need to be replaced. I think this qualifies for a follow-up PR doing just that. This one is just part of code that got moved around …